tests/test_importer.py

```diff
@@ -69,39 +74,118 @@ def test_number_of_imported_entities(entity_name, Model, data_import, data_folde
 
 
 @pytest.mark.django_db
-def test_sources(data_import, data_folder):
-    """
-    All records in the source data should have their own sources. Check all
-    attributes of all entities and confirm that each has its own source.
-    """
-    src_related_attrs = [attr for attr in dir(AccessPoint.objects.first())
-                         if attr.endswith('_related')]
-    for access_point in AccessPoint.objects.all():
+def test_all_data_points_have_sources(data_import):
+    def check_for_sources(complex_field):
+        unsourced_models = (OrganizationRealStart, OrganizationOpenEnded)
+        value = entity_attr.get_value()
+
+        if value:
+            if any(isinstance(value, Model) for Model in unsourced_models):
+                assert not entity_attr.get_sources()
+            else:
+                try:
+                    assert entity_attr.get_sources()
+                except AssertionError:
+                    return value
+
+    errors = []
+
+    for Model in [Person, Organization, Violation]:
+        for entity in Model.objects.all():
+            for entity_attr in getattr(entity, 'complex_fields', []):
+                error = check_for_sources(entity_attr)
+                if error:
+                    errors.append(error)
+
+            for entity_attrs in getattr(entity, 'complex_lists', []):
+                for entity_attr in entity_attrs.get_list():
+                    error = check_for_sources(entity_attr)
+                    if error:
+                        errors.append(error)
+
+    if errors:
+        raise Exception('The following data points are unsourced:\n{}'.format(errors))
+
+
+@pytest.mark.django_db
+def test_sources_only_created_for_data_points_they_evidence(data_import, data_folder):
+    '''
+    In https://github.com/security-force-monitor/sfm-cms/issues/637, we
+    discovered a bug in which sources were multiplied when derived from the
+    original, mutable source list, rather than a shallow copy. Test data was
+    created such that we could test sources weren't inadvertantly assigned
+    to unrelated attributes. This test asserts that extraneous sources are not
+    created, based on that data.
+
+    N.b., there are not constraints on what data points can share sources
+    outside of this test.
+    '''
+    sourced_attributes = [
+        attr for attr in dir(AccessPoint.objects.first())
+        if attr.endswith('_related')
+    ]
+
+    access_points_for_test = Q()
+
+    for substring in ('alpha', 'beta', 'gamma', 'delta', 'is-member', 'has-member'):
+        access_points_for_test |= Q(source__title__icontains=substring)
+
+    for access_point in AccessPoint.objects.filter(access_points_for_test):
```

Comment:
The new test data does not conform with the pattern of source creation we used when we wrote the test. Explicitly query for the sources created for the purpose of this test, so new test data may be added without breaking this test.